### PR TITLE
Be nc news 8

### PR DESCRIPTION
**Pull request by dancancode85**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -236,3 +236,102 @@ describe("postComment", () => {
...
+        article_id: 5,
+        title: expect.any(String),
+        body: expect.any(String),
+        votes: 0,
+        topic: expect.any(String),
+        author: expect.any(String),
+        created_at: expect.any(String),
+      });
+    });
+  });
```
Comment (aloush-dev) [inline __tests__/integration.test.js]: This should really be a 400 Bad Request since this endpoint is expecting an object.

Comment (dancancode85) [reply __tests__/integration.test.js]: Cool!! Will do that now

File: models/models.js [hunk 2]
```diff
@@ -105,11 +105,33 @@ const postNewComment = (article_id, author, body) => {
...
+const patchArticleVotesByID = (id, inc_votes) => {
+  if (inc_votes === undefined) {
+    return db
+      .query(`SELECT * FROM articles WHERE article_id = $1;`, [id])
+      .then((article) => {
+        if (!article.rows.length) {
+          return Promise.reject({ status: 404, msg: "Not found" });
+        }
+        return article.rows[0]; 
+      });
```
Comment (aloush-dev) [inline models/models.js]: If the inc_votes is undefined thats when we want to sent a 400 bad request.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see a test for.
200 - If the object has extra properties it ignores them and updates the article.

A test which I left a comment for about a bad request, also should apply if the inc_votes is not in the object thats sent.

Please resubmit once added.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see a test for.
200 - If the object has extra properties it ignores them and updates the article.

A test which I left a comment for about a bad request, also should apply if the inc_votes is not in the object thats sent.

Please resubmit once added.

File: models/models.js [hunk 3]
```diff
@@ -105,11 +105,35 @@ const postNewComment = (article_id, author, body) => {
...
+const patchArticleVotesByID = (id, inc_votes) => {
+  if (inc_votes === undefined) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1;`, [id])
+    .then((article) => {
+      if (!article.rows.length) {
+        return Promise.reject({ status: 404, msg: "Not found" });
+      }
```
Comment (aloush-dev) [inline models/models.js:117]: This is fine but you do have a similar model you've already written which does this exact functionality.

**Review by aloush-dev — APPROVED** [review]
Awesome looking good now.

Merge when ready

**Review by aloush-dev — APPROVED** [review]
Awesome looking good now.

Merge when ready

**Pull request merged**